src/Form/createForm.tsx

```diff
@@ -0,0 +1,300 @@
+            });
+          }
+        });
+        return;
+      }
+      Keyboard.dismiss();
+    };
+
+    private clearAdjustContentOffsetTimeout() {
+      if (this.adjustContentOffsetTimeoutToken != null) {
+        clearTimeout(this.adjustContentOffsetTimeoutToken);
+        this.adjustContentOffsetTimeoutToken = null;
+      }
+    }
+
+    private scrollToFocusedTextInput() {
+      this.clearAdjustContentOffsetTimeout();
+      const currentlyFocusedNodeId = TextInput.State.currentlyFocusedField();
+      if (currentlyFocusedNodeId == null) {
+        return;
+      }
+      this.scrollToFocusedField(currentlyFocusedNodeId);
+    }
+
+    private scrollToFocusedField(nodeID: number) {
+      this.clearAdjustContentOffsetTimeout();
+
+      if (this.scrollViewRef.current == null) {
+        return;
+      }
+
+      const scrollViewNodeID = findNodeHandle(this.scrollViewRef.current);
+      if (scrollViewNodeID == null) {
+        return;
+      }
+
+      this.clearAdjustContentOffsetTimeout();
+
+      (UIManager as any).viewIsDescendantOf(
+        nodeID,
+        scrollViewNodeID,
+        (isAncestor: boolean) => {
+          if (!isAncestor) {
+            return;
+          }
+          UIManager.measureLayout(
+            nodeID,
+            scrollViewNodeID,
+            () => {},
+            (_x, y) => {
+              const { scrollViewContentHeight, scrollViewHeight } = this.state;
+              if (scrollViewContentHeight <= scrollViewHeight) {
+                return;
+              }
+              const maxContentOffsetY =
+                this.state.scrollViewContentHeight -
+                this.state.scrollViewHeight;
+              const desiredContentOffsetY = Math.min(
+                y < 100 ? 0 : (y * 2) / 3,
+                maxContentOffsetY
+              );
+              if (this.scrollViewRef.current) {
+                this.scrollViewRef.current.scrollTo({
+                  x: 0,
+                  y: desiredContentOffsetY,
+                  animated: true,
+                });
+              }
+            }
+          );
+        }
+      );
+    }
+
+    onScrollViewContentSizeChange = (
+      contentWidth: number,
+      contentHeight: number
+    ) => {
+      this.setState({
+        scrollViewContentHeight: contentHeight,
+      });
+      if (this.props.onContentSizeChange) {
+        this.props.onContentSizeChange(contentWidth, contentHeight);
+      }
+    };
+
+    onScrollViewLayout = (e: LayoutChangeEvent) => {
+      this.setState({
+        scrollViewHeight: e.nativeEvent.layout.height,
+        containerMinHeight: e.nativeEvent.layout.height,
+      });
+      if (this.props.onLayout) {
+        this.props.onLayout(e);
+      }
+    };
+
+    render() {
+      const { children, ...restProps } = this.props;
+
+      return (
+        <Provider value={this.state.context}>
+          <ScrollView
+            ref={this.scrollViewRef}
+            alwaysBounceVertical={false}
```

Comment:
Any reason for this? It can be overridden by restProps.

Reply:
The purpose of this is I dont want the form can be scrolled if the content size is smaller than the scrollview size. Imagine there is only 1 input field and if the form can be scrolled, I think that is weird UX.

Reply:
It would be great if you can put this rationale in comment.